src/components/TaskDetail.js

```diff
@@ -0,0 +1,128 @@
+import React, { useState, useEffect } from 'react';
+import MDEditor from '@uiw/react-md-editor';
+import { useAuth } from '../contexts/AuthContext';
+import { useTheme } from '../contexts/ThemeContext';
+import rehypeSanitize from 'rehype-sanitize';
+
+export function TaskDetail({ task, isOpen, onClose, onUpdate, onDelete }) {
+  const { user } = useAuth();
+  const { isDark } = useTheme();
+  const [description, setDescription] = useState(task?.description || '');
+  const [isEditing, setIsEditing] = useState(false);
+  const [showDeleteConfirm, setShowDeleteConfirm] = useState(false);
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
const [showDeleteConfirm, setShowDeleteConfirm] = useState(false);

useEffect(() => {
  return () => setShowDeleteConfirm(false);
}, [isOpen, task]);
```
Add a useEffect cleanup to reset showDeleteConfirm when the modal closes or task changes, preventing stale state if the modal is reopened.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;